### DISCUSSION TITLE
drlg@home

### DISCUSSION THREAD

**Discussion by vlolteanu** [diasurgical/DevilutionX#2185] [Ideas]
Remember seti@home? They were looking for aliens. Most of us probably care more about DIablo than about aliens.

There's been some talk about seeds that are problematic for level generation (these are a few instances that I'm aware of: https://github.com/diasurgical/devilutionX/pull/2115#discussion_r652633549, https://github.com/diasurgical/devilutionX/pull/1629). There are "only" 2^32 possible game layouts and some of them are bad.

I think you know where this is going.

It should be possible to try out all possible seeds and see if they work. A non-exhaustive list of deal-breakers would be:
 * crashing the app,
 * non-traversability of a level,
 * unreachable spots,
 * no walls at the edges.

Seeds that cause problems can then be blacklisted.

I'm not suggesting doing a full-blown volunteer computing app using BOINC or something similar; we could divide the load among us on a voluntary basis. I also don't plan on coding this anytime soon, so if anyone wants to take the idea and run with it, feel free.

**Comment by AJenbo** [comment]
generating the levels take about a week, validating them would probably add as much

**Comment by ephphatha** [comment]
I'm planning to look at the level generation code as part of reviewing the use of the existing RNG, so I'll put some updates here if I find anything interesting.

Pretty sure there's 2^31+1 potential layouts as well due to the use of abs() in the existing implementation, that almost halves the search space 😄

**Comment by mewmew** [comment]
> I'm planning to look at the level generation code as part of reviewing the use of the existing RNG, so I'll put some updates here if I find anything interesting.

I'd love to learn about your drlg deep dive @ephphatha! How did it go? Did you stumble on anything curious or weird? Any broken seeds that were unknown since before?

Cheerful regards,
Robin

**Reply by ephphatha** [reply]
I have done nothing at scale, been looking at other cleanups. When making the RNG PR I confirmed it behaved the same as vanilla code and didn't go any further with the deep dive 😅 .
